### PR TITLE
common/gpu/intel: add extraPackages32

### PR DESCRIPTION
**Pull request by OrichalcumCosmonaut**
###### Description of changes

`common/gpu/intel` doesn’t add the matching 32-bit packages for video acceleration, resulting in (at least) Steam being unable to play videos on store pages. This can be diagnosed with <kbd>Help</kbd> → <kbd>Steam Runtime Diagnostics</kbd>:

```
"x11/vdpau" : {
  "messages" : [
    "Failed to open VDPAU backend libvdpau_va_gl.so: cannot open shared object file: No such file or directory",
    "vdp_device_create_x11 (display, screen, &device, &vdp_get_proc_address) failed: 1"
  ],
  "renderer" : null,
  "version" : null,
  "issues" : [
    "cannot-draw"
  ],
  "exit-status" : 1
},
"x11/vaapi" : {
  "messages" : [
    "libva info: VA-API version 1.21.0",
    "libva info: Trying to open /run/opengl-driver-32/lib/dri/iHD_drv_video.so",
    "libva info: Trying to open /usr/lib/dri/iHD_drv_video.so",
    "libva error: dlopen of /usr/lib/dri/iHD_drv_video.so failed: /usr/lib/dri/iHD_drv_video.so: wrong ELF class: ELFCLASS64",
    "libva info: Trying to open /usr/lib32/dri/iHD_drv_video.so",
    "libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so",
    "libva info: Trying to open /usr/lib/i386-linux-gnu/dri/iHD_drv_video.so",
    "libva info: va_openDriver() returns -1",
    "libva info: Trying to open /run/opengl-driver-32/lib/dri/i965_drv_video.so",
    "libva info: Trying to open /usr/lib/dri/i965_drv_video.so",
    "libva error: dlopen of /usr/lib/dri/i965_drv_video.so failed: /usr/lib/dri/i965_drv_video.so: wrong ELF class: ELFCLASS64",
    "libva info: Trying to open /usr/lib32/dri/i965_drv_video.so",
    "libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so",
    "libva info: Trying to open /usr/lib/i386-linux-gnu/dri/i965_drv_video.so",
    "libva info: va_openDriver() returns -1",
    "vaInitialize (va_display, &major_version, &minor_version) failed: unknown libva error (-1)"
  ],
  "renderer" : null,
  "version" : null,
  "issues" : [
    "cannot-draw"
  ],
  "exit-status" : 1
},
```

This change fixes this, at least for Intel GPUs:

```
"x11/vdpau" : {
  "messages" : [
    "libva info: VA-API version 1.21.0",
    "libva info: Trying to open /run/opengl-driver-32/lib/dri/iHD_drv_video.so",
    "libva info: Found init function __vaDriverInit_1_21",
    "libva info: va_openDriver() returns 0"
  ],
  "renderer" : "OpenGL/VAAPI backend for VDPAU\n",
  "version" : null
},
"x11/vaapi" : {
  "messages" : [
    "libva info: VA-API version 1.21.0",
    "libva info: Trying to open /run/opengl-driver-32/lib/dri/iHD_drv_video.so",
    "libva info: Found init function __vaDriverInit_1_21",
    "libva info: va_openDriver() returns 0",
    "Testing H264Main decoding",
    "vaEndPicture (va_display, context) failed: invalid parameter (18)",
    "Testing MPEG2Simple decoding",
    "vaEndPicture (va_display, context) failed: invalid parameter (18)",
    "Testing post-processing with VAProfileNone",
    "vaCreateBuffer (va_display, context, VAEncMiscParameterBufferType, sizeof (VAEncMiscParameterBuffer) + sizeof (VAEncMiscParameterBufferQualityLevel), 1, NULL, &misc_buf_id) failed: invalid parameter (18)"
  ],
  "renderer" : null,
  "version" : null,
  "issues" : [
    "cannot-draw"
  ],
  "exit-status" : 1
},
```

The VA-API diagnostic does still seem to fail (although it fails differently?), but I can confirm videos on Steam store pages work with this change.

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: common/gpu/intel/default.nix [hunk 1]
```diff
@@ -21,6 +21,12 @@
         libvdpau-va-gl
         intel-media-driver
       ];
+
+      hardware.opengl.extraPackages32 = with pkgs.driversi686Linux; [
+        (if (lib.versionOlder (lib.versions.majorMinor lib.version) "23.11") then vaapiIntel else intel-vaapi-driver)
+        libvdpau-va-gl
```
Comment (SuperSandro2000) [inline common/gpu/intel/default.nix:27]: Do you have an idea in which scenarios libvdpau-va-gl is useful?

Comment (OrichalcumCosmonaut) [reply common/gpu/intel/default.nix:27]: Not sure.  I just copied the list of packages already being added to `.extraPackages`.

Comment (Mic92) [reply common/gpu/intel/default.nix:27]: I guess it's fine because it's only used if you have `opengl.driSupport32Bit` enabled?

Comment (Mic92) [reply common/gpu/intel/default.nix:27]: We can still remove it, if it causes problems down the line.

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**